examples/summarize_rlhf/reward_model/train_reward_model_gptj.py

```diff
@@ -0,0 +1,148 @@
+        pair = {}
+        prompt = sample["prompt"]
+        chosen_summary = sample["chosen"]
+        rejected_summary = sample["rejected"]
+        if chosen_summary == rejected_summary:
+            continue
+        if len(chosen_summary.split()) < 5 or len(rejected_summary.split()) < 5:
+            continue
+        pair["chosen"] = prompt + "\n" + chosen_summary
+        pair["rejected"] = prompt + "\n" + rejected_summary
+        pairs.append(pair)
+    return pairs
+
+
+class PairwiseDataset(Dataset):
+    def __init__(self, pairs, tokenizer, max_length):
+        self.chosen_input_ids = []
+        self.chosen_attn_masks = []
+        self.rejected_input_ids = []
+        self.rejected_attn_masks = []
+        for pair in tqdm(pairs):
+            chosen, rejected = pair["chosen"], pair["rejected"]
+            chosen_encodings_dict = tokenizer(
+                "<|startoftext|>" + chosen + "<|endoftext|>",
+                truncation=True,
+                max_length=max_length,
+                padding="max_length",
+                return_tensors="pt",
+            )
+            rejected_encodings_dict = tokenizer(
+                "<|startoftext|>" + rejected + "<|endoftext|>",
+                truncation=True,
+                max_length=max_length,
+                padding="max_length",
+                return_tensors="pt",
+            )
+            self.chosen_input_ids.append(chosen_encodings_dict["input_ids"])
+            self.chosen_attn_masks.append(chosen_encodings_dict["attention_mask"])
+            self.rejected_input_ids.append(rejected_encodings_dict["input_ids"])
+            self.rejected_attn_masks.append(rejected_encodings_dict["attention_mask"])
+
+    def __len__(self):
+        return len(self.chosen_input_ids)
+
+    def __getitem__(self, idx):
+        return (
+            self.chosen_input_ids[idx],
+            self.chosen_attn_masks[idx],
+            self.rejected_input_ids[idx],
+            self.rejected_attn_masks[idx],
+        )
+
+
+class DataCollatorReward:
+    def __call__(self, data):
+        batch = {}
+        batch["input_ids"] = torch.cat([f[0] for f in data] + [f[2] for f in data])
+        batch["attention_mask"] = torch.cat([f[1] for f in data] + [f[3] for f in data])
+        batch["labels"] = torch.tensor([0] * len(data) + [1] * len(data))
+        return batch
+
+
+def compute_metrics(eval_preds):
+    chosen_end_scores = eval_preds.predictions[0]  # chosen scores
+    rejected_end_scores = eval_preds.predictions[1]  # rejected scores
+
+    result = {}
+    acc = sum(chosen_end_scores > rejected_end_scores) / len(rejected_end_scores)
+    result["accuracy"] = acc
+
+    return result
+
+
+if __name__ == "__main__":
+    tokenizer = AutoTokenizer.from_pretrained("EleutherAI/gpt-j-6B")
+    tokenizer.pad_token = tokenizer.eos_token
+
+    if not os.path.exists("rm_checkpoint"):
+        os.mkdir("rm_checkpoint")
+
+    training_args = TrainingArguments(
```

Comment:
Can you not just use a config file for this?

Reply:
> Note that this is the SFT training script that is transformers.Trainer-based. I don't believe transformers supplies a config-to-[transformers.TrainingArguments](https://huggingface.co/docs/transformers/main_classes/trainer#transformers.TrainingArguments) converter, so I'm okay leaving this for now.

See the related comment in the conversation below.